fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/FieldValue.java

```diff
@@ -320,13 +324,57 @@ public static Optional<FieldPath> stripFieldPrefixMaybe(@Nonnull FieldPath field
         final var potentialPrefixFieldOrdinals = potentialPrefixPath.getFieldOrdinals();
         final var potentialPrefixFieldTypes = potentialPrefixPath.getFieldTypes();
         for (int i = 0; i < potentialPrefixPath.size(); i++) {
-            if (fieldPathOrdinals.get(i) != potentialPrefixFieldOrdinals.get(i) || !fieldPathTypes.get(i).equals(potentialPrefixFieldTypes.get(i))) {
+            if (fieldPathOrdinals.get(i) != potentialPrefixFieldOrdinals.get(i) || !equalIgnoringFieldNames(fieldPathTypes.get(i), potentialPrefixFieldTypes.get(i), new HashSet<>())) {
                 return Optional.empty();
             }
         }
         return Optional.of(fieldPath.subList(potentialPrefixPath.size(), fieldPath.size()));
     }
 
+    @SuppressWarnings("PMD.CompareObjectsWithEquals") // Deliberate use of reference equality
+    private static boolean equalIgnoringFieldNames(@Nonnull Type t1, @Nonnull Type t2, @Nonnull Set<Pair<Type, Type>> explored) {
```

Comment:
This method needs some comments and should adhere to other styling like `final` in the signature, etc.

Other than that I am wondering if this method can be simplified/restructured into the field path logic.

Reply:
We discussed this during an in-person review, and we now think the right thing is to remove the type check entirely from `stripFieldPrefixMaybe`. After that, this method becomes unnecessary, so I've removed it.